### DISCUSSION TITLE
Need help installed but get this error when trying to train

### DISCUSSION THREAD

**Discussion by BotLifeGaming** [bmaltais/kohya_ss#1012] [Q&A]
Been at this all day

===================================BUG REPORT===================================
Welcome to bitsandbytes. For bug reports, please submit your error trace to: https://github.com/TimDettmers/bitsandbytes/issues
For effortless bug reporting copy-paste your error into this form: https://docs.google.com/forms/d/e/1FAIpQLScPB8emS3Thkp66nvqwmjTEgxp8Y9ufuWTzFyr9kJ5AoI47dQ/viewform?usp=sf_link
================================================================================
CUDA SETUP: Loading binary C:\Users\GamingPC.DESKTOP-UE5QGUL\Desktop\Lora\kohya_ss\venv\lib\site-packages\bitsandbytes\libbitsandbytes_cuda116.dll...
use 8-bit AdamW optimizer | {}
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ C:\Users\GamingPC.DESKTOP-UE5QGUL\Desktop\Lora\kohya_ss\train_db.py:486 in <module>              │
│                                                                                                  │
│   483 │   args = parser.parse_args()                                                             │
│   484 │   args = train_util.read_config_from_file(args, parser)                                  │
│   485 │                                                                                          │
│ ❱ 486 │   train(args)                                                                            │
│   487                                                                                            │
│                                                                                                  │
│ C:\Users\GamingPC.DESKTOP-UE5QGUL\Desktop\Lora\kohya_ss\train_db.py:206 in train                 │
│                                                                                                  │
│   203 │                                                                                          │
│   204 │   # acceleratorがなんかよろしくやってくれるらしい                                        │
│   205 │   if train_text_encoder:                                                                 │
│ ❱ 206 │   │   unet, text_encoder, optimizer, train_dataloader, lr_scheduler = accelerator.prep   │
│   207 │   │   │   unet, text_encoder, optimizer, train_dataloader, lr_scheduler                  │
│   208 │   │   )                                                                                  │
│   209 │   else:                                                                                  │
│                                                                                                  │
│ C:\Users\GamingPC.DESKTOP-UE5QGUL\Desktop\Lora\kohya_ss\venv\lib\site-packages\accelerate\accele │
│ rator.py:1143 in prepare                                                                         │
│                                                                                                  │
│   1140 │   │   elif self.distributed_type == DistributedType.MEGATRON_LM:                        │
│   1141 │   │   │   result = self._prepare_megatron_lm(*args)                                     │
│   1142 │   │   else:                                                                             │
│ ❱ 1143 │   │   │   result = tuple(                                                               │
│   1144 │   │   │   │   self._prepare_one(obj, first_pass=True, device_placement=d) for obj, d i  │
│   1145 │   │   │   )                                                                             │
│   1146 │   │   │   result = tuple(self._prepare_one(obj, device_placement=d) for obj, d in zip(  │
│                                                                                                  │
│ C:\Users\GamingPC.DESKTOP-UE5QGUL\Desktop\Lora\kohya_ss\venv\lib\site-packages\accelerate\accele │
│ rator.py:1144 in <genexpr>                                                                       │
│                                                                                                  │
│   1141 │   │   │   result = self._prepare_megatron_lm(*args)                                     │
│   1142 │   │   else:                                                                             │
│   1143 │   │   │   result = tuple(                                                               │
│ ❱ 1144 │   │   │   │   self._prepare_one(obj, first_pass=True, device_placement=d) for obj, d i  │
│   1145 │   │   │   )                                                                             │
│   1146 │   │   │   result = tuple(self._prepare_one(obj, device_placement=d) for obj, d in zip(  │
│   1147                                                                                           │
│                                                                                                  │
│ C:\Users\GamingPC.DESKTOP-UE5QGUL\Desktop\Lora\kohya_ss\venv\lib\site-packages\accelerate\accele │
│ rator.py:995 in _prepare_one                                                                     │
│                                                                                                  │
│    992 │   │   │   if isinstance(obj, torch.utils.data.DataLoader):                              │
│    993 │   │   │   │   return self.prepare_data_loader(obj, device_placement=device_placement)   │
│    994 │   │   │   elif isinstance(obj, torch.nn.Module):                                        │
│ ❱  995 │   │   │   │   return self.prepare_model(obj, device_placement=device_placement)         │
│    996 │   │   │   elif isinstance(obj, torch.optim.Optimizer):                                  │
│    997 │   │   │   │   optimizer = self.prepare_optimizer(obj, device_placement=device_placemen  │
│    998 │   │   │   │   return optimizer                                                          │
│                                                                                                  │
│ C:\Users\GamingPC.DESKTOP-UE5QGUL\Desktop\Lora\kohya_ss\venv\lib\site-packages\accelerate\accele │
│ rator.py:1290 in prepare_model                                                                   │
│                                                                                                  │
│   1287 │   │   if self.state.dynamo_plugin.backend != DynamoBackend.NO:                          │
│   1288 │   │   │   if not is_torch_version(">=", "2.0"):                                         │
│   1289 │   │   │   │   raise ValueError("Using `torch.compile` requires PyTorch 2.0 or higher."  │
│ ❱ 1290 │   │   │   model = torch.compile(model, **self.state.dynamo_plugin.to_kwargs())          │
│   1291 │   │   return model                                                                      │
│   1292 │                                                                                         │
│   1293 │   def _prepare_deepspeed(self, *args):                                                  │
│                                                                                                  │
│ C:\Users\GamingPC.DESKTOP-UE5QGUL\Desktop\Lora\kohya_ss\venv\lib\site-packages\torch\__init__.py │
│ :1441 in compile                                                                                 │
│                                                                                                  │
│   1438 │   │   mode = "default"                                                                  │
│   1439 │   if backend == "inductor":                                                             │
│   1440 │   │   backend = _TorchCompileInductorWrapper(mode, options, dynamic)                    │
│ ❱ 1441 │   return torch._dynamo.optimize(backend=backend, nopython=fullgraph, dynamic=dynamic,   │
│   1442                                                                                           │
│   1443                                                                                           │
│   1444 def _register_device_module(device_type, module):                                         │
│                                                                                                  │
│ C:\Users\GamingPC.DESKTOP-UE5QGUL\Desktop\Lora\kohya_ss\venv\lib\site-packages\torch\_dynamo\eva │
│ l_frame.py:413 in optimize                                                                       │
│                                                                                                  │
│   410 │   │   def toy_example(a, b):                                                             │
│   411 │   │   │   ...                                                                            │
│   412 │   """                                                                                    │
│ ❱ 413 │   check_if_dynamo_supported()                                                            │
│   414 │   # Note: The hooks object could be global instead of passed around, *however* that wo   │
│   415 │   # for a confusing API usage and plumbing story wherein we nest multiple .optimize ca   │
│   416 │   # There is some prior art around this, w/r/t nesting backend calls are enforced to b   │
│                                                                                                  │
│ C:\Users\GamingPC.DESKTOP-UE5QGUL\Desktop\Lora\kohya_ss\venv\lib\site-packages\torch\_dynamo\eva │
│ l_frame.py:375 in check_if_dynamo_supported                                                      │
│                                                                                                  │
│   372                                                                                            │
│   373 def check_if_dynamo_supported():                                                           │
│   374 │   if sys.platform == "win32":                                                            │
│ ❱ 375 │   │   raise RuntimeError("Windows not yet supported for torch.compile")                  │
│   376 │   if sys.version_info >= (3, 11):                                                        │
│   377 │   │   raise RuntimeError("Python 3.11+ not yet supported for torch.compile")             │
│   378                                                                                            │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
RuntimeError: Windows not yet supported for torch.compile
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ C:\Users\GamingPC.DESKTOP-UE5QGUL\AppData\Local\Programs\Python\Python310\lib\runpy.py:196 in    │
│ _run_module_as_main                                                                              │
│                                                                                                  │
│   193 │   main_globals = sys.modules["__main__"].__dict__                                        │
│   194 │   if alter_argv:                                                                         │
│   195 │   │   sys.argv[0] = mod_spec.origin                                                      │
│ ❱ 196 │   return _run_code(code, main_globals, None,                                             │
│   197 │   │   │   │   │    "__main__", mod_spec)                                                 │
│   198                                                                                            │
│   199 def run_module(mod_name, init_globals=None,                                                │
│                                                                                                  │
│ C:\Users\GamingPC.DESKTOP-UE5QGUL\AppData\Local\Programs\Python\Python310\lib\runpy.py:86 in     │
│ _run_code                                                                                        │
│                                                                                                  │
│    83 │   │   │   │   │      __loader__ = loader,                                                │
│    84 │   │   │   │   │      __package__ = pkg_name,                                             │
│    85 │   │   │   │   │      __spec__ = mod_spec)                                                │
│ ❱  86 │   exec(code, run_globals)                                                                │
│    87 │   return run_globals                                                                     │
│    88                                                                                            │
│    89 def _run_module_code(code, init_globals=None,                                              │
│                                                                                                  │
│ in <module>:7                                                                                    │
│                                                                                                  │
│   4 from accelerate.commands.accelerate_cli import main                                          │
│   5 if __name__ == '__main__':                                                                   │
│   6 │   sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])                         │
│ ❱ 7 │   sys.exit(main())                                                                         │
│   8                                                                                              │
│                                                                                                  │
│ C:\Users\GamingPC.DESKTOP-UE5QGUL\Desktop\Lora\kohya_ss\venv\lib\site-packages\accelerate\comman │
│ ds\accelerate_cli.py:45 in main                                                                  │
│                                                                                                  │
│   42 │   │   exit(1)                                                                             │
│   43 │                                                                                           │
│   44 │   # Run                                                                                   │
│ ❱ 45 │   args.func(args)                                                                         │
│   46                                                                                             │
│   47                                                                                             │
│   48 if __name__ == "__main__":                                                                  │
│                                                                                                  │
│ C:\Users\GamingPC.DESKTOP-UE5QGUL\Desktop\Lora\kohya_ss\venv\lib\site-packages\accelerate\comman │
│ ds\launch.py:918 in launch_command                                                               │
│                                                                                                  │
│   915 │   elif defaults is not None and defaults.compute_environment == ComputeEnvironment.AMA   │
│   916 │   │   sagemaker_launcher(defaults, args)                                                 │
│   917 │   else:                                                                                  │
│ ❱ 918 │   │   simple_launcher(args)                                                              │
│   919                                                                                            │
│   920                                                                                            │
│   921 def main():                                                                                │
│                                                                                                  │
│ C:\Users\GamingPC.DESKTOP-UE5QGUL\Desktop\Lora\kohya_ss\venv\lib\site-packages\accelerate\comman │
│ ds\launch.py:580 in simple_launcher                                                              │
│                                                                                                  │
│   577 │   process.wait()                                                                         │
│   578 │   if process.returncode != 0:                                                            │
│   579 │   │   if not args.quiet:                                                                 │
│ ❱ 580 │   │   │   raise subprocess.CalledProcessError(returncode=process.returncode, cmd=cmd)    │
│   581 │   │   else:                                                                              │
│   582 │   │   │   sys.exit(1)                                                                    │
│   583                                                                                            │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
CalledProcessError: Command '['C:\\Users\\GamingPC.DESKTOP-UE5QGUL\\Desktop\\Lora\\kohya_ss\\venv\\Scripts\\python.exe','train_db.py', '--enable_bucket', '--pretrained_model_name_or_path=C:/Users/GamingPC.DESKTOP-UE5QGUL/Desktop/SUPER SD
2.0/stable-diffusion-webui/models/Stable-diffusion/v1-5-pruned.ckpt',
'--train_data_dir=C:/Users/GamingPC.DESKTOP-UE5QGUL/Desktop/Lora/Lora_training_data/jkabak/image',
'--resolution=512,512', '--output_dir=C:/Users/GamingPC.DESKTOP-UE5QGUL/Desktop/Lora/Lora_training_data/jkabak/model',
'--logging_dir=C:/Users/GamingPC.DESKTOP-UE5QGUL/Desktop/Lora/Lora_training_data/jkabak/log',
'--save_model_as=safetensors', '--output_name=last', '--max_data_loader_n_workers=0', '--learning_rate=1e-05',
'--lr_scheduler=cosine', '--lr_warmup_steps=60', '--train_batch_size=1', '--max_train_steps=600',
'--save_every_n_epochs=1', '--mixed_precision=fp16', '--save_precision=fp16', '--cache_latents',
'--optimizer_type=AdamW8bit', '--max_data_loader_n_workers=0', '--bucket_reso_steps=64', '--xformers',
'--bucket_no_upscale']' returned non-zero exit status 1.

**Comment by J-Kazusa** [comment]
+1

**Comment by JetsprintRacer** [comment]
I got similar error, when tried to train SDXL LoRA, using a checkpoint from Civitai. Fixed this by downloading an SDXL base model.

**Comment by ronbere** [comment]
> I got similar error, when tried to train SDXL LoRA, using a checkpoint from Civitai. Fixed this by downloading an SDXL base model.

same error for me
